src/schema-parser/raml-schema-parser.ts

```diff
@@ -0,0 +1,69 @@
+import { SchemaParser, ParseSchemaInput, ValidateSchemaInput } from "../schema-parser";
+import type { AsyncAPISchema, SchemaValidateResult } from '../types';
+import yaml from 'js-yaml';
+import * as lib from "webapi-parser";
+const wap = lib.WebApiParser;
+const r2j = require('ramldt2jsonschema');
+
+export function RamlSchemaParser(): SchemaParser {
+    return {
+      validate,
+      parse,
+      getMimeTypes,
+    }
+  }
+
+async function parse(input: ParseSchemaInput<unknown, unknown>): Promise<AsyncAPISchema> {
+    const message = (input.meta as any).message
+    try {
+        const payload = formatPayload(input.data);
+
+        // Draft 6 is compatible with 7.
+        const jsonModel = await r2j.dt2js(payload, 'tmpType', { draft: '06' });
+        const convertedType = jsonModel.definitions.tmpType;
+
+        message['x-parser-original-schema-format'] = input.schemaFormat || input.defaultSchemaFormat;
+        message['x-parser-original-payload'] = payload;
+        message.payload = convertedType;
+        delete message.schemaFormat;
+    } catch (e) {
+        console.error(e);
+    }
+
+    return message.payload;
+}
+
+function getMimeTypes() {
+  return [
+    'application/raml+yaml;version=1.0',
+  ];
+}
+
+async function validate(input: ValidateSchemaInput<unknown, unknown>): Promise<SchemaValidateResult[]> {
+  const payload = formatPayload(input.data);
+  const parsed = await wap.raml10.parse(payload);
+  const report = await wap.raml10.validate(parsed);
+  if (report.conforms) {
+    // No errors found.
+    return [];
+  }
+  
+  let validateResult: SchemaValidateResult[] = [];
+  report.results.forEach(result => {
+    validateResult.push({
+      message: result.message,
+      path: input.path, // Info provided by the RAML parser doesn't provide a real path to the error.
```

Comment:
In the custom parsers path of single error should point to from perspective of given payload - in other words, payload should be treated as root of error, we will concatenate error path in one place, concatenate path of schema + path of error, so here you should return empty array.

 ```suggestion
      path: [] // Info provided by the RAML parser doesn't provide a real path to the error.
```

Reply:
Let's say `input.path` value is: `["channels", "myChannel", "publish", "message", "payload"]`, the behaviour will be different that https://github.com/asyncapi/parser-js/pull/578/files#r952869403

Reply:
But there you operate on error results, in mentioned comment you create mock data for testing. `.path` of input is location on which data you operate some logic, `error.path` is path of the error, but from `.path` point of view - it's a local path that we will merge after validation 😄

Reply:
That means, this is also wrong, right? https://github.com/asyncapi/parser-js/blob/next-major/src/schema-parser/asyncapi-schema-parser.ts#L68

Reply:
Changed!